### PR TITLE
new module to change admin pwd through ssh

### PR DESCRIPTION
**Pull request by ivanbojer**
ISSUE TYPE

New Module Pull Request
COMPONENT NAME

panos_admpwd

ANSIBLE VERSION

ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
SUMMARY

New panos module that allows for user password changes given ssh with ssh_key as authentication method. This is used in cloud environments where box comes up w/o password and with ssh_key only.

### CODE REVIEW

**Comment by ivanbojer** [comment]
Migrated from: https://github.com/ansible/ansible-modules-extras/pull/3258

**Comment by ivanbojer** [comment]
ready_for_review

File: lib/ansible/modules/network/panos/panos_admpwd.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        username=dict(default='admin'),
+        key_filename=dict(),
+        newpassword=dict(no_log=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+    if not HAS_LIB:
+        module.fail_json(msg='paramiko is required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_admpwd.py:188]: Set required=true in the argument_spec

```ip_address=dict(required=True)```

File: lib/ansible/modules/network/panos/panos_admpwd.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        module.fail_json(msg="ip_address should be specified")
+    key_filename = module.params["key_filename"]
+    if not key_filename:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_admpwd.py:191]: Same as ```ip_address```

File: lib/ansible/modules/network/panos/panos_admpwd.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        module.fail_json(msg="key_filename should be specified")
+    newpassword = module.params["newpassword"]
+    if not newpassword:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_admpwd.py:194]: Same as ```ip_address```

**Review by privateip — CHANGES_REQUESTED** [review]
Ansible 2.3 now requires a snippet of metadata for every new module.  Please add the following to your module and then it can be merged.

```
ANSIBLE_METADATA = {'status': ['preview'],
                    'supported_by': 'community',
                    'version': '1.0'}
```

Details on the usage of metadata can be found [here](https://github.com/ansible/proposals/issues/30) for the time being while documentation is still being updated.

**Comment by ivanbojer** [comment]
@privateip Thank you for the review. I've incorporated all changes.

**Comment by ivanbojer** [comment]
ready_for_review

**Review by privateip — APPROVED** [review]

**Pull request merged**